internal/pkg/term/progress/deploy.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package progress
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/deploy"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/term/color"
+)
+
+// ResourceMatcher is a function that returns true if the resource event matches a criteria.
+type ResourceMatcher func(deploy.Resource) bool
+
+// HumanizeResourceEvents groups raw deploy events under human-friendly progress texts that can be passed into the Events() method.
```

Comment:
Nit: Can we wrap this at a narrow tab width?

Reply:
Done!